### DISCUSSION TITLE
Code review by Will McGugan

### DISCUSSION THREAD

**Discussion by C4ptainCrunch** [ics-py/ics-py#299] [Ideas]
Hi 👋 

I know i'm not really active over here anymore (sorry about that 😕 ) but i stumbled upon [this thread](https://twitter.com/willmcgugan/status/1439862215675453448) by @willmcgugan, the creator of [Rich](https://github.com/willmcgugan/rich) proposing to do code reviews for open-source projets.

What would you think of asking a review of our codebase ? I think that benefiting from a strong and well designed API would be really beneficial for ics-py. Because let's be honest, the API i designed when starting the project is probably not really good 🙈 . Also designing a good API to handle recurring events will be quite a big task i guess.

**Comment by N-Coder** [comment]
I think that would be awesome! The everyday usage of v0.8 in terms of API for parsing, creating, and serializing Calendars should be pretty much done and fix by now, so that would already be ready for review. Furthermore, I'm not quite decided yet how I should do the two remaining points mentioned [here](https://github.com/ics-py/ics-py/issues/245#issuecomment-923798362), so getting some input and discussion on the advanced usecases of customizing the parser by tweaking the handling of non-standard inputs and adding custom properties would be nice.

**Reply by C4ptainCrunch** [reply]
Great :) Would you like to be the one to post the issue/request on [the repo](https://github.com/willmcgugan/codereviews/issues) or should i do it ? Or do you want to wait ?

**Reply by N-Coder** [reply]
I submitted an issue [here](https://github.com/willmcgugan/codereviews/issues/25)

**Comment by make-github-pseudonymous-again** [comment]
@C4ptainCrunch  I am all for a code review but I would let @N-Coder decide what's best in terms of timing. If he wants to finish what he is working on first I would prefer to postpone any additional outside help. I have never contributed much to this code so I am also fine if my advice is ignored.
